proxyd/server.go

```diff
@@ -197,22 +213,37 @@ func (s *Server) HandleRPC(w http.ResponseWriter, r *http.Request) {
 
 	origin := r.Header.Get("Origin")
 	userAgent := r.Header.Get("User-Agent")
-	exemptOrigin := s.limExemptOrigins[strings.ToLower(origin)]
-	exemptUserAgent := s.limExemptUserAgents[strings.ToLower(userAgent)]
 	// Use XFF in context since it will automatically be replaced by the remote IP
 	xff := stripXFF(GetXForwardedFor(ctx))
-	var ok bool
-	if exemptOrigin || exemptUserAgent {
-		ok = true
-	} else {
-		if xff == "" {
-			log.Warn("rejecting request without XFF or remote IP")
-			ok = false
+	isUnlimitedOrigin := s.isUnlimitedOrigin(origin)
+	isUnlimitedUserAgent := s.isUnlimitedUserAgent(userAgent)
+
+	if xff == "" {
+		writeRPCError(ctx, w, nil, ErrInvalidRequest("request does not include a remote IP"))
+		return
+	}
+
+	isLimited := func(method string) bool {
+		if isUnlimitedOrigin || isUnlimitedUserAgent {
+			return false
+		}
+
+		var lim limiter.Store
+		if method == "" {
+			lim = s.mainLim
 		} else {
-			_, _, _, ok, _ = s.lim.Take(ctx, xff)
+			lim = s.overrideLims[method]
+		}
+
+		if lim == nil {
+			return false
 		}
+
+		_, _, _, ok, _ := lim.Take(ctx, xff)
+		return !ok
 	}
-	if !ok {
+
+	if isLimited("") {
```

Comment:
Is the empty string here correct?

Reply:
This is saying that if the non method override is limited correct?

Reply:
Yes - this selects the global limiter.